substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/methodhandles/MethodHandleFeature.java

```diff
@@ -121,7 +122,10 @@ public void duringSetup(DuringSetupAccess access) {
         memberNameIsMethod = ReflectionUtil.lookupMethod(memberNameClass, "isMethod");
         memberNameIsConstructor = ReflectionUtil.lookupMethod(memberNameClass, "isConstructor");
         memberNameIsField = ReflectionUtil.lookupMethod(memberNameClass, "isField");
-        memberNameGetParameterTypes = ReflectionUtil.lookupMethod(memberNameClass, "getParameterTypes");
+        memberNameGetMethodType = ReflectionUtil.lookupMethod(memberNameClass, "getMethodType");
+
+        Class<?> methodTypeClass = access.findClassByName("java.lang.invoke.MethodType");
```

Comment:
why is this happening reflectively? `MethodType` and `parameterArray` are public

Reply:
My bad, copy pasted code without giving it much thought.